sentry/src/main/java/io/sentry/SentryEnvelopeItem.java

```diff
@@ -148,6 +149,50 @@ public static SentryEnvelopeItem fromUserFeedback(
     return new SentryEnvelopeItem(itemHeader, cachedItem::getBytes);
   }
 
+  public static SentryEnvelopeItem fromAttachment(
+      @NotNull ILogger logger, final @NotNull Attachment attachment) {
+
+    final CachedItem cachedItem =
+        new CachedItem(
+            () -> {
+              if (attachment.getBytes() != null) {
+                return attachment.getBytes();
+              } else if (attachment.getPath() != null) {
+                ByteArrayOutputStream outputStream;
+
+                try (FileInputStream stream = new FileInputStream(attachment.getPath())) {
+                  byte[] bytes = new byte[1024];
+                  outputStream = new ByteArrayOutputStream();
+                  int length;
+                  int offset = 0;
+                  while ((length = stream.read(bytes)) != -1) {
+                    outputStream.write(bytes, offset, length);
+                  }
+                } catch (IOException | SecurityException exception) {
+                  logger.log(
+                      SentryLevel.ERROR,
+                      exception,
+                      "Serializing attachment %s failed.",
+                      attachment.getFilename());
+                  outputStream = new ByteArrayOutputStream();
+                }
+
+                return outputStream.toByteArray();
```

Comment:
M: a few things here concerns me, if one of the `write` throws, we still return a portion of it with `outputStream.toByteArray()`, not sure if this is a good idea, I am all in for either the whole thing or nothing.

`ByteArrayOutputStream` could be created inside of the `try-with-resources` so it's gonna be deallocated/closed automatically.

Might be possible to wrap the `FileInputStream` with `BufferedInputStream`, usually Buffered solutions would work better on Mobile

Reply:
Oh my! Was I out of my mind 🙈? Thanks for the important input. I'm going to change this.

Where do you have the info from, that buffered solutions work better on mobile?

Reply:
It's not only on mobile but as we have limited resources, it's more important on mobile.
official docs already mention a bit about it https://docs.oracle.com/javase/tutorial/essential/io/buffers.html